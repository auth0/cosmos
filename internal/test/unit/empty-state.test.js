import React from 'react'
import { shallow, mount } from 'enzyme'
import { EmptyState, Icon, Heading } from '@auth0/cosmos'

const emptyStateFactory = ({
  title = 'Some title',
  icon = 'copy',
  link = '/some-place',
  action = { label: 'Something', icon: 'copy', handler: () => {} }
} = {}) => (
  <EmptyState title={title} icon={icon} action={action} link={link}>
    Some description
  </EmptyState>
)

describe('Empty State', () => {
  it('renders properly', () => {
    const wrapper = shallow(emptyStateFactory())
    expect(wrapper).toMatchSnapshot()
  })

  it('renders properly without help link', () => {
    const wrapper = shallow(emptyStateFactory({ link: null }))
    expect(wrapper).toMatchSnapshot()
  })

  it('renders properly without text', () => {
    const wrapper = shallow(emptyStateFactory({ text: null }))
    expect(wrapper).toMatchSnapshot()
  })

  it('renders the provided action and the helper is being called', () => {
    const action = { title: 'Something', icon: 'copy', handler: jest.fn() }
    const wrapper = mount(emptyStateFactory({ action }))

    expect(wrapper.exists('button')).toBe(true)

    const button = wrapper.find('button')
    button.simulate('click')

    expect(action.handler).toHaveBeenCalledTimes(1)
  })
})
