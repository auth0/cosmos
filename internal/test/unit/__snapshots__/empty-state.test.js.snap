// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty State renders properly 1`] = `
<styled.div
  data-cosmos-key="empty-state"
>
  <Styled(Heading)
    size={1}
  >
    Some title
  </Styled(Heading)>
  <styled.div>
    <Icon
      color="blue"
      name="copy"
      size={110}
    />
    <FreeText
      useParagraph={true}
    >
      Some description
    </FreeText>
    <styled.div>
      <Link
        href="/some-place"
        target="_blank"
      >
        Learn More 
        <i>
           
        </i>
      </Link>
    </styled.div>
  </styled.div>
  <Button
    appearance="cta"
    disabled={false}
    icon="copy"
    iconAlign="left"
    loading={false}
    onClick={[Function]}
    size="large"
    success={false}
  >
    Something
  </Button>
</styled.div>
`;

exports[`Empty State renders properly without help link 1`] = `
<styled.div
  data-cosmos-key="empty-state"
>
  <Styled(Heading)
    size={1}
  >
    Some title
  </Styled(Heading)>
  <styled.div>
    <Icon
      color="blue"
      name="copy"
      size={110}
    />
    <FreeText
      useParagraph={true}
    >
      Some description
    </FreeText>
  </styled.div>
  <Button
    appearance="cta"
    disabled={false}
    icon="copy"
    iconAlign="left"
    loading={false}
    onClick={[Function]}
    size="large"
    success={false}
  >
    Something
  </Button>
</styled.div>
`;

exports[`Empty State renders properly without text 1`] = `
<styled.div
  data-cosmos-key="empty-state"
>
  <Styled(Heading)
    size={1}
  >
    Some title
  </Styled(Heading)>
  <styled.div>
    <Icon
      color="blue"
      name="copy"
      size={110}
    />
    <FreeText
      useParagraph={true}
    >
      Some description
    </FreeText>
    <styled.div>
      <Link
        href="/some-place"
        target="_blank"
      >
        Learn More 
        <i>
           
        </i>
      </Link>
    </styled.div>
  </styled.div>
  <Button
    appearance="cta"
    disabled={false}
    icon="copy"
    iconAlign="left"
    loading={false}
    onClick={[Function]}
    size="large"
    success={false}
  >
    Something
  </Button>
</styled.div>
`;
